src/main/java/it/gov/pagopa/payhub/auth/service/user/registration/UserRegistrationService.java

```diff
@@ -12,14 +15,26 @@ public class UserRegistrationService {
     private final ExternalUserIdObfuscatorService externalUserIdObfuscatorService;
     private final FiscalCodeObfuscatorService fiscalCodeObfuscatorService;
     private final UsersRepository usersRepository;
+    private final MyPayUsersService myPayUsersService;
+    private final MyPivotUsersService myPivotUsersService;
 
-    public UserRegistrationService(ExternalUserIdObfuscatorService externalUserIdObfuscatorService, FiscalCodeObfuscatorService fiscalCodeObfuscatorService, UsersRepository usersRepository) {
+    public UserRegistrationService(ExternalUserIdObfuscatorService externalUserIdObfuscatorService,
+        FiscalCodeObfuscatorService fiscalCodeObfuscatorService, UsersRepository usersRepository,
+        MyPayUsersService myPayUsersService, MyPivotUsersService myPivotUsersService) {
         this.externalUserIdObfuscatorService = externalUserIdObfuscatorService;
         this.fiscalCodeObfuscatorService = fiscalCodeObfuscatorService;
         this.usersRepository = usersRepository;
+        this.myPayUsersService = myPayUsersService;
+        this.myPivotUsersService = myPivotUsersService;
     }
 
     public User registerUser(String externalUserId, String fiscalCode, String iamIssuer, String firstName, String lastName, String email){
+        myPayUsersService.registerMyPayUser(externalUserId, fiscalCode, firstName, lastName, email);
+        log.info("Registering user on MyPay having mappedExternalUserId {}", externalUserId);
+
+        myPivotUsersService.registerMyPivotUser(externalUserId, fiscalCode, firstName, lastName, email);
```

Comment:
externalUserId is not mappedExternalUserId!
Move these statements after buildUser and user user.getMappedExternalUserId instead